### DISCUSSION TITLE
Support for Grove MGS V2 Sensor needed

### DISCUSSION THREAD

**Discussion by Joylancer** [arendst/Tasmota#22462] [Ideas]
MGS V1 is supportet, MGS V2 not.
New Sensor: https://wiki.seeedstudio.com/Grove-Multichannel-Gas-Sensor-V2/
i2c: 0x08
is it possible?

**Comment by spockatieee** [comment]
It's supported under ESPhome. Maybe port the code over, if that makes sense?

**Reply by s-hadinger** [reply]
This should be easy with Berry code on esp32. Or adapt the current driver (I haven't looked at it)

**Reply by spockatieee** [reply]
Either approach is fine. I'm happy to test the code anyone produces.

**Reply by spockatieee** [reply]
Here is the code for ESPhome, for reference. [https://github.com/wifwucite/esphome-grove-multi-gas-v2](url) and also from the ESPhome repo [https://api-docs.esphome.io/grove__gas__mc__v2_8cpp_source](url)

**Reply by s-hadinger** [reply]
On hold, will look at it in a few weeks
